src/register_manager.hh

```diff
@@ -116,6 +116,8 @@ public:
     }
 };
 
+using RegisterMap = HashMap<Codepoint, std::unique_ptr<Register>, MemoryDomain::Registers>;
```

Comment:
I think this should be nested inside RegisterManager, although it should not be necessary anymore if you use auto deduction for begin/end as suggested in the previous comment.